core/common-util/src/test/java/uk/gov/gchq/gaffer/commonutil/iterable/LimitedCloseableIterableTest.java

```diff
@@ -99,4 +100,35 @@ public void shouldThrowExceptionIfStartIsBiggerThanEnd() {
             assertNotNull(e.getMessage());
         }
     }
+
+    @Test
+    public void shouldThrowExceptionIfDataIsTruncated() {
```

Comment:
Please could you add another test, when limit is equal to the length of the iterable and truncate is false.